Elbow Saver/Models/ExerciseTimerViewModel.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  ExerciseTimerViewModel.swift
+//  ExerciseTimerViewModel
+//
+//  Created by Daniel Luo on 8/23/21.
+//
+
+import Foundation
+
+class ExerciseTimerViewModel: ObservableObject {
+    
+    // Fixed values, not user-adjustable
+    /// The number of eccentric repetitions per set of exercise.
+    static let repsPerSet: Int = 3
+    /// The duration over which each rep is to be performed.
+    static let secondsPerRep: Int = 4
+    /// The duration between reps during which user returns to the exercise start position
+    static let secondsBetweenReps: Int = 1
+    
+    /// Settings for session including total number of sets, rest period between sets.
+    let sessionSettings: ExerciseSessionSettings
+    
+    /**
+     Initialize a new exercise timer, guiding the user through exercise sesssion.
+     Call `startSession()` to begin timing the session.
+     
+     - Parameters:
+        - sessionSettings: A struct containing settings for the exercise session. 
+     */
+    init(sessionSettings: ExerciseSessionSettings) {
+        self.sessionSettings = sessionSettings
+        secondsRemainingInRestPeriod = sessionSettings.restPeriodInSeconds
+    }
+    
+    /// The current set of exercise session
+    @Published private(set) var currentSet: Int = 1
+    /// The current repetition in the current set
+    @Published private(set) var currentRep: Int = 1
+    /// The number of seconds remaining for performance of each rep.
+    @Published private(set) var secondsRemainingForRep = secondsPerRep
+    /// The number of seconds remaining that the user has to return to the exercise start position
+    @Published private(set) var secondsRemainingBetweenReps = secondsBetweenReps
+    /// The number of seconds remaining for the rest period
+    @Published private(set) var secondsRemainingInRestPeriod: Int
+    
+    /// The  possible states of the exercise timer.
+    enum TimerState {
+        /// Timer is not running.
+        case stopped
+        /// Timer is counting time for a rep.
+        case performingRep
+        /// Timer is counting time in between two reps.
+        case betweenReps
+        /// Timer is counting time in between two sets.
+        case betweenSets
+    }
+    @Published private(set) var currentTimerState: TimerState = .stopped
+    
+    private var timer: Timer?
+    private var frequency: TimeInterval = 1.0
+    private var setsCompleted = 0
+    
+    
+    /// Start the exercise session and timer
+    func startSession() {
+        guard setsCompleted < sessionSettings.totalNumberOfSets else { return }
+        currentTimerState = .performingRep
+        
+        timer = Timer.scheduledTimer(withTimeInterval: frequency, repeats: true) { [weak self] timer in
+            if let self = self {
```

Comment:
super minor nit but this could be `guard let self = self else { return }` to save yourself an indent and a line. 

I used to call these out in actual work PRs but mostly ignore them nowadays since the functionality is fine.

buuuuuuut since its you and I'm super caffeinated....... ;)